packages/astro/src/runtime/server/render/component.ts

```diff
@@ -239,12 +239,14 @@ If you're still stuck, please open an issue on GitHub or join us at https://astr
 	// This is a custom element without a renderer. Because of that, render it
 	// as a string and the user is responsible for adding a script tag for the component definition.
 	if (!html && typeof Component === 'string') {
+		// Sanitize tag name because some people might try to inject attributes 🙄
+		const Tag = Component.trim().split(/\s+/)[0].trim();
```

Comment:
Can we do an `indexOf(' ')` first to check that the tag name contains a space? This code looks a little expensive so would be good for well-intended code.

Reply:
I switched this to a regex check because I realized there are other cases than whitespace.